### ISSUE TITLE
sha-1 vs sha1

### ISSUE THREAD

**Issue by ignatenkobrain** [hyperium/headers#38]
Hello,

I'm packaging `headers` for Fedora and I noticed that it uses `sha-1` which depends on quite some number of crates which had last release in 2016/2017 and we try to avoid packaging such crates. Would it be feasible to swith to `sha1`?

Thanks in advance!

**Comment by seanmonstar** [comment]
The `sha-1` crate is published by the "RustCrypto" team, which I find valuable.

**Comment by trevyn** [comment]
@seanmonstar Possible to bump `sha-1` to latest, then? https://github.com/hyperium/headers/pull/72

**Comment by barrowsys** [comment]
commit 31fe3e19e5763b95c9b5dfed1569bef555380dc4 replaces `sha-1` with `sha1`

at the moment, the released `0.3.7` version of the crate does not have this change, and does not build.
```
   Compiling sha-1 v0.10.3
   Compiling headers v0.3.7 (/home/barrow/src/headers)
error[E0463]: can't find crate for `sha1`
  --> src/lib.rs:83:1
   |
83 | extern crate sha1;
   | ^^^^^^^^^^^^^^^^^^ can't find crate

For more information about this error, try `rustc --explain E0463`.
error: could not compile `headers` due to previous error
```

this is causing many downstream crates to also not build. a new minor release should fix.

**Comment by seanmonstar** [comment]
> many downstream crates to also not build

This sounds like a different issue, there's no reason headers v0.3.7 should suddenly stop working.

Wrapping up here, though, we've changed to the sha1 crate as of v0.3.8.
